doc/concepts/configuration.rst

```diff
@@ -0,0 +1,651 @@
+
+The example below shows a sample configuration of a single Tarantool instance:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/instance_scope/config.yaml
+    :language: yaml
+    :dedent:
+
+-   The ``instances`` section includes only one instance named *instance001*.
+    The ``iproto.listen`` option sets a port used to listen for incoming requests.
+-   The ``replicasets`` section contains one replica set named *replicaset001*.
+-   The ``groups`` section contains one group named *group001*.
+
+
+.. _configuration_scopes:
+
+Configuration scopes
+********************
+
+This section shows how to control a scope the specified configuration option is applied to.
+Most of the configuration options can be applied to a specific instance, replica set, group, or to all instances globally.
+
+-   *Instance*
+
+    To apply specific configuration options to a concrete instance,
+    specify such options for this instance only.
+    In the example below, ``iproto.listen`` is applied to *instance001* only.
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/instance_scope/config.yaml
+        :language: yaml
+        :emphasize-lines: 7-8
+        :dedent:
+
+-   *Replica set*
+
+    In this example, ``iproto.listen`` is in effect for all instances in *replicaset001*.
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/replicaset_scope/config.yaml
+        :language: yaml
+        :emphasize-lines: 5-6
+        :dedent:
+
+-   *Group*
+
+    In this example, ``iproto.listen`` is in effect for all instances in *group001*.
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/group_scope/config.yaml
+        :language: yaml
+        :emphasize-lines: 3-4
+        :dedent:
+
+-   *Global*
+
+    In this example, ``iproto.listen`` is applied to all instances of all groups.
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/global_scope/config.yaml
+        :language: yaml
+        :emphasize-lines: 1-2
+        :dedent:
+
+
+.. NOTE::
+
+    The :ref:`Configuration reference <configuration_reference>` contains information to which scopes each configuration option can be applied.
+
+
+.. _configuration_replica_set_scopes:
+
+Configuration scopes: Replica set example
+*****************************************
+
+The example below shows how specific configuration options work in different configuration scopes for a replica set with a manual failover.
+You can learn more about configuring replication from :ref:`Replication tutorials <how-to-replication>`.
+
+..  literalinclude:: /code_snippets/snippets/replication/instances.enabled/manual_leader/config.yaml
+    :language: yaml
+    :emphasize-lines: 1-8,10-12,14-15,21,24-25,27-28,30-31
+    :end-before: Load sample data
+    :dedent:
+
+-   ``credentials`` (*global*)
+
+    Options in this section grant the specified roles to the *replicator* and *client* users.
```

Comment:
Are these users created automatically? Does this require actions from the reader?

Reply:
Changed to:

> This section is used to create the *replicator* and *client* users and assign them the specified roles.